x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/edit_package_policy_page/index.tsx

```diff
@@ -76,16 +85,19 @@ export const EditPackagePolicyPage: React.FunctionComponent = () => {
     GetOnePackagePolicyResponse['item']
   >();
 
+  const policyId = agentPolicy?.id ?? '';
+
   // Retrieve agent policy, package, and package policy info
   useEffect(() => {
     const getData = async () => {
       setIsLoadingData(true);
       setLoadingError(undefined);
       try {
-        const [{ data: agentPolicyData }, { data: packagePolicyData }] = await Promise.all([
-          sendGetOneAgentPolicy(policyId),
-          sendGetOnePackagePolicy(packagePolicyId),
-        ]);
+        const { data: packagePolicyData } = await sendGetOnePackagePolicy(packagePolicyId);
+        const { data: agentPolicyData } = await sendGetOneAgentPolicy(
+          packagePolicyData?.item.policy_id ?? 'id-missing-in-package-policy'
```

Comment:
this seems odd. Not a big fan of putting.. log-like traces into values. 

you also don't _technically_ know that that could not be a valid future policy ID. 

You should check the existence of that property and value before calling sendGetOneAgentPolicy.
You're in a `try` already, could check it, and throw if it's missing (if that's the appropriate behavior).

```ts
if (!item.policy_id) { throw new Error("ID missing in package policy") }
```

Or if it's something you move past, by grabbing a default, perhaps something like

```ts
const { data: agentPolicyData } = item.policy_id ? await sendGetOneAgentPolicy() : someDefault()
```

Or if having the value itself is optional 

```ts
const { data?: agentPolicyData } = item.policy_id ? await sendGetOneAgentPolicy() : null
```

something along those lines seems more intentional than sending intentionally fake data down the line to return.. who knows what. An error? A null? Accidentally a real object?

Reply:
I agree with @pzl here we can probably validate that packagePolicyData is not null and the request succeed before sending this request

Reply:
Also you probably want to check the `error` field of the response too, if a request fail we probably want to throw that error

Reply:
Yeah, I meant to comment on this and why I did it.
`throw`ing here would not be good because the entire UI will likely crash. instead, I wanted to piggy back off the error handling in place for failed API calls. If this did happen to error, I wanted to have something that can tell us (or support team) some info about where to go look.

That being said, you just pointed out that all of this is wrapped in a `try {}` block (why did I not realize that? 🤦‍♂️ ), so I as you suggest, doing a check and then explicitly `throw new Error()` will be a better approach.

Thanks for the feedback. making the change now

Reply:
Thanks @nchaulet . Good point on processing each individual API call checking for errors - in the prior code both of those would be handled (use of `Promise.all()`) but with my change they are not. Will fix